chain/client/src/info.rs

```diff
@@ -177,6 +181,45 @@ impl InfoHelper {
         }
     }
 
+    fn record_epoch_settlement_info(head: &Tip, client: &crate::client::Client) {
```

Comment:
Any idea how long this takes? For 43200 heights in an epoch I just wanna make sure this won't take too long and make validators miss block/chunk production, since I assume this is run on ClientActor.

Reply:
I ran perf on the RPC node and the function does not show up in the graph. I even looked for `log_summary` call. I believe they are negligible.

The function is also called once per epoch to avoid unnecessary computation.

Check the CPU flamegraph below.
![perf](https://user-images.githubusercontent.com/8384611/227204316-a3553eee-9971-438b-af7d-7a9253c98a69.svg)

Reply:
I have created a localnet with 4 nodes to run perf during the epoch transition. The results are [here](https://drive.google.com/drive/folders/1Ett7y00K49iswqd8GGX2aaqpHAMkz_59?usp=share_link) but I cannot see a difference. 

Let me know if you have any suggestion on how to test this more accurately.